### PR TITLE
feat(routerStore): add v8 migration schematic

### PR DESCRIPTION
**Pull request by timdeschryver**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/ngrx/platform/blob/master/CONTRIBUTING.md#commit
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Documentation has been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

```
[ ] Bugfix
[x] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Other... Please describe:
```

## What is the current behavior?

Closes #

## What is the new behavior?

Add a schematic for the breaking change introduced in #1672.
Replace `StoreRouterConnectingModule` with `StoreRouterConnectingModule.forRoot()`.

## Does this PR introduce a breaking change?

```
[ ] Yes
[x] No
```

## Other information

Extract `createReplaceChange` and `createChangeRecorder` to schematics-core and refactor the store schematics.

### CODE REVIEW

**Comment by brandonroberts** [comment]
Will there be additional instructions needed for running these migrations? Something like

```
ng update @ngrx/store --from 7 --to 8
```

**Comment by timdeschryver** [comment]
It should be sufficient with `ng update @ngrx/store`.
I'm not 100% sure of it and wanted to try it if we have a beta version.
I ran it via `verdaccio` locally, it picks up the migration file but it doesn't seem to execute the migration - I'm not sure if this is because of `verdaccio` or not since the schematic does work during the tests.

**Comment by brandonroberts** [comment]
Hopefully we can cut a beta next week and test it with that.

**Comment by brandonroberts** [comment]
TIL: https://github.com/angular/angular/pull/29735

File: modules/router-store/migrations/migration.json [hunk 1]
```diff
@@ -6,6 +6,11 @@
       "description": "The road to v6",
       "version": "5.2",
       "factory": "./6_0_0/index"
+    },
+    "ngrx-router-store-migration-02": {
+      "description": "The road to v8",
+      "version": "7.0",
```
Comment (brandonroberts) [inline modules/router-store/migrations/migration.json]: Should this be `8-beta` also?

Comment (timdeschryver) [reply modules/router-store/migrations/migration.json]: Yes, I updated the wrong one. `ngrx/store` is modified in #1716

Comment (timdeschryver) [reply modules/router-store/migrations/migration.json]: Modified in 29249e

**Pull request merged**